### ISSUE TITLE
Document vSphere certificate verification support

### ISSUE THREAD

**Issue by stuclem** [vmware/vic#2570]
Per PR https://github.com/vmware/vic/pull/2554. Don't see an associated issue.

**Comment by stuclem** [comment]
PR https://github.com/vmware/vic/pull/2554 is merged.

**Comment by stuclem** [comment]
Updated the following topics with thumbprint info:

- [Virtual Container Host Deployment Options](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_installation/vch_installer_options.md#thumbprint)
- []()
- []()
- []()
- []()

The updates are in https://github.com/stuclem/vic/commit/435459ed1db4125220a566ef8314c0055188be04.

**Comment by dougm** [comment]
lgtm @stuclem 

The thumbprint is only required if the VC/ESX certificate is not signed by a known/trusted CA.

When using a known/trusted CA, updating the VC/ESX certificate will not impact VIC.
When using thumbprint based verification, updating the VC/ESX certificate will prevent VIC from managing existing container VMs, creating new ones, etc.  However, existing container VMs will continue to run.  User will need to use vic-machine to update.

A couple of cli options to get the thumbprint ahead of time:

```
govc about.cert -k -u $target -thumbprint
```

```
openssl s_client -connect $target </dev/null 2>/dev/null | openssl x509 -fingerprint -noout
```

**Comment by stuclem** [comment]
Updated the doc as follows:

-----

### `thumbprint` ###

Short name: None

The thumbprint of the vCenter Server or ESXi host certificate. Specify this option if your vSphere environment uses untrusted, self-signed certificates. If your vSphere environment uses trusted certificates that are signed by a known Certificate Authority (CA), you do not need to specify the `--thumbprint` option.

**NOTE** If your vSphere environment uses untrusted, self-signed certificates, you can run `vic-machine create` without the `--thumbprint` option by using the `--force` option. However, running `vic-machine create` with the `--force` option rather than providing the certificate thumbprint is not recommended, because it permits man-in-the-middle attacks to go undetected.

To obtain the thumbprint of the vCenter Server or ESXi host certificate, run `vic-machine create` without the specifying the `--thumbprint` or `--force` options. The deployment of the virtual container host fails, but the resulting error message includes the required certificate thumbprint. 

<pre>Failed to verify certificate for target=<i>vcenter_or_esxi_host</i> (thumbprint=<i>thumbprint</i>)
</pre>

You can copy the thumbprint from the error message and run vic-machine create again, including the `thumbprint` option.

<pre>--thumbprint <i>certificate_thumbprint</i></pre>

-----

@dougm regarding your upfront methods of getting the cert thumbprint, I couldn't get them to work. Where there are several ways of doing things, our editorial guidelines recommend only documenting one method, if possible. So, I'll stick with running `vic-machine create` twice, as I think that this is the easiest way.

I will address how to manage a change in certificate when I get to documenting upgrade in https://github.com/vmware/vic/issues/843. 

Updated the following files with the new info about trusted/untrusted certs:

- [Virtual Container Host Deployment Options](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_installation/vch_installer_options.md#thumbprint)
- [Examples of Deploying a Virtual Container Host](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_installation/vch_installer_examples.md)
- [vSphere Integrated Containers Engine Security Overview](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_installation/security.md)
- [List Virtual Container Hosts and Obtain Their IDs](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/list_vch.md)
- [Virtual Container Host List Options](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/list_vch_options.md#thumbprint)
- [Obtain Information About a Virtual Container Host](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/inspect_vch.md)
- [Virtual Container Host Inspect Options](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/inspect_vch_options.md#thumbprint)
- [Delete a Virtual Container Host](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/remove_vch.md)
- [Virtual Container Host Delete Options](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/delete_vch_options.md#thumbprint)

The updates are in https://github.com/stuclem/vic/commit/b2bde73e921240b928f99be4b135c7700d35d5db.

**Comment by dougm** [comment]
@stuclem I suppose openssl didn't work on windows.  govc about.cert does require a new version, just released 0.10.0 today: https://github.com/vmware/govmomi/releases/tag/v0.10.0
